synapse/replication/tcp/redis.py

```diff
@@ -228,3 +228,44 @@ def buildProtocol(self, addr):
         p.password = self.password
 
         return p
+
+
+def lazyConnection(
+    reactor,
+    host="localhost",
+    port=6379,
+    dbid=None,
+    reconnect=True,
+    charset="utf-8",
+    password=None,
+    connectTimeout=None,
+    replyTimeout=None,
+    convertNumbers=True,
+):
+    """Equivalent to `txredisapi.lazyConnection`, except allows specifying a
+    reactor.
+    """
+
+    isLazy = True
+    poolsize = 1
+
+    uuid = "%s:%d" % (host, port)
+    factory = txredisapi.RedisFactory(
+        uuid,
+        dbid,
+        poolsize,
+        isLazy,
+        txredisapi.ConnectionHandler,
+        charset,
+        password,
+        replyTimeout,
+        convertNumbers,
+    )
+    factory.continueTrying = reconnect
+    for x in range(poolsize):
+        reactor.connectTCP(host, port, factory, connectTimeout)
+
+    if isLazy:
+        return factory.handler
+    else:
+        return factory.deferred
```

Comment:
This is a copy and paste from the function in `txredisapi` fwiw, but with `reactor` given as a parameter.